### PR TITLE
Add spotless code formatter

### LINKED ISSUE #6605: Migrate Ktlint plugin to Kotlinter
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same feature request.
- [X] I will describe the problem with as much detail as possible.
- [X] This issue only contains a request for one single feature, **not** multiple (related) features.

### App version

v6.503

### Problem you are trying to solve

The `org.jlleitschuh.gradle.ktlint` plugin is outdated, can't support new version Ktlint, we should migrate to `org.jmailen.kotlinter`.

### Suggested solution

https://github.com/jeremymailen/kotlinter-gradle

### Screenshots / Drawings / Technical details

_No response_

### PR DESCRIPTION
**Pull request by wmontwe**
This changes the ktlint plugin to the [Spotless](https://github.com/diffplug/spotless) [Gradle plugin](https://github.com/diffplug/spotless/tree/main/plugin-gradle) with ktlint support. Spotless also allows to run other formatters, so this comes with Markdown, KTS and misc rules configured.

This spares the Java formatter due to it's massive changeset. A simple rule matching most of current Java code style is:

```
spotless {
    java {
        googleJavaFormat().aosp()
        target("**/*.java")
        targetExclude("plugins/openpgp-api-lib/", "ui-utils/")
    }
}
```

Due to some issues with ktlint 0.48.x it is still using 0.47.1. Problems expected to be fixed in ktlint 0.49.0
- [Formatting enum class](https://github.com/pinterest/ktlint/issues/1786)
- [Formatting with no-semi rule removes semicolons in enum](https://github.com/pinterest/ktlint/issues/1733)

Closes #6605

### CODE REVIEW

**Review by cketti — APPROVED** [review]
`pull_request_template.md` needs to be updated. Otherwise LGTM :+1:

**Review by cketti — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -94,10 +94,35 @@ allprojects {
...
     }
+}
 
-    apply plugin: 'org.jlleitschuh.gradle.ktlint'
-    ktlint {
-        version = libs.versions.ktlint.get()
+spotless {
+    kotlin {
+        ktlint(libs.versions.ktlint.get())
+            .setEditorConfigPath("$projectDir/.editorconfig")
```
Comment (Goooler) [inline build.gradle:102]: Unnecessary override, it's the default.

File: build.gradle [hunk 2]
```diff
@@ -94,10 +94,35 @@ allprojects {
...
+        target("**/*.kt")
+        targetExclude("**/build/", "**/resources/", "plugins/openpgp-api-lib/")
+        endWithNewline()
```
Comment (Goooler) [inline build.gradle:105]: This rule had been covered by `insert_final_newline` in `.editorconfig`.

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -38,7 +39,7 @@ ksp = "com.google.devtools.ksp:1.8.0-1.0.8"
 kotlin-android = { id = "org.jetbrains.kotlin.android", version.ref = "kotlin" }
 kotlin-parcelize = { id = "org.jetbrains.kotlin.plugin.parcelize", version.ref = "kotlin" }
 kotlin-jvm = { id = "org.jetbrains.kotlin.jvm", version.ref = "kotlin" }
-ktlint = "org.jlleitschuh.gradle.ktlint:11.0.0"
+spotless = { id = "com.diffplug.spotless", version.ref = "spotless" }
```
Comment (Goooler) [inline gradle/libs.versions.toml:42]: Just inline this version.

File: build.gradle (hunk 1 above)
Comment (wmontwe) [reply build.gradle:102]: Right, I'll create a follow up change

File: build.gradle (hunk 2 above)
Comment (wmontwe) [reply build.gradle:105]: Right, I'll create a follow up change

**Pull request merged**